### PR TITLE
fix: npm prune & add pr verification checks

### LINKED ISSUE #3275: GitHub action to verify autogenerated doc content for pull requests
## Description
When running `make release` during our release process, we ran into an issue where a step was missing from the docs and a pull request was merged. This caused an `Err: unpruned files` error.

## Exit Criteria
- run GH action on pull request for branch `release-next`
- checkout PR branch & install dependencies
- run `touch lib/utils/config/definitions; npm run postsnap`
- check if `git status` is clean
- if dir has changes we fail, otherwise we pass

### PR DESCRIPTION
**Pull request by gijun19**
## Details
Release `v7.14.0` gave us an error with docs related to `npm-pack`. This PR includes a patch where we ran `make docs` and the changes.

Adds "PR Validation Checks" workflow for pull requests on branches `release-next` and `latest`. 

1. `pr-title-format` will check if the PR title follows conventional commits formatting
2. `clean-working-dir` will check if we have a clean working directory after forcing a regeneration of docs and running `make prune`

## References
Closes #3275

### CODE REVIEW

**Comment by gijun19** [comment]
https://github.com/gimli01/cli/pull/2

As per documentation, for this to work across forks the workflow needs to exist on the main branch. So this needs to get merged into `npm/cli` before workflows work properly. The docs for the action can be found [here](https://github.com/marketplace/actions/semantic-pull-request)

I posted a link to the PR from my pr branch into my fork. you can see the checks passed successfully there. we're just changing the event trigger to be `pull_request_target` from `pull_request` for this to work for PRs coming from forks.

File: package-lock.json [hunk 1]
```diff
@@ -7373,7 +7373,135 @@
...
+        "tap-yaml",
+        "to-fast-properties",
+        "type-fest",
+        "unicode-length",
+        "widest-line",
+        "wrap-ansi",
+        "wrappy",
+        "yallist",
+        "yaml",
+        "yoga-layout-prebuilt"
```
Comment (ruyadorno) [inline package-lock.json]: this is completely unrelated to the change, can you just `git reset origin/release-next package-lock.json` please?

**Review by ruyadorno — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-validation-checks.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+on:
+  # pull_request:
+  #   types:
+  #     - opened
+  #     - edited
+  #     - synchronize
+  #   branches:
+  #     - release-next
+  #     - latest
```
Comment (darcyclarke) [inline .github/workflows/pr-validation-checks.yml:11]: Any reason to keep this commented out code/config?

Comment (gijun19) [reply .github/workflows/pr-validation-checks.yml:11]: useful when someone's developing against this workflow from a fork to test the workflow locally but other than that no.
